src/i2p.cpp

```diff
@@ -410,7 +410,7 @@ void Session::Disconnect()
             Log("Destroying session %s", m_session_id);
         }
     }
-    m_control_sock->Reset();
+    m_control_sock = std::make_unique<Sock>(INVALID_SOCKET);
```

Comment:
Last nit: could this be `m_control_sock.release()`, or does the smart pointer always need to point to something?

Reply:
In the current code the smart pointer always has to point to something because we unconditionally dereference it in a few places.

However, that change is coming in https://github.com/bitcoin/bitcoin/pull/21878, from commit `i2p: avoid using Sock::Get() for checking for a valid socket` - it will use an empty unique_ptr to denote that there is no valid socket and together with a few other commits will remove the `Sock::Get()` method so that the value of the underlying socket file descriptor will be completely hidden inside the `Sock` class.

Reply:
Great, so you already thought of that, thanks!